go/oasis-test-runner/oasis/args.go

```diff
@@ -598,20 +656,60 @@ func (args *argBuilder) byzantineVersionFakeEnclaveID(rt *Runtime) *argBuilder {
 		MrEnclave: *rt.mrEnclaves[0],
 		MrSigner:  *rt.mrSigner,
 	}
-	args.vec = append(args.vec, "--"+byzantine.CfgVersionFakeEnclaveID, eid.String())
+	args.vec = append(args.vec, argument{false, []string{"--" + byzantine.CfgVersionFakeEnclaveID, eid.String()}})
 	return args
 }
 
 func (args *argBuilder) byzantineActivationEpoch(epoch beacon.EpochTime) *argBuilder {
-	args.vec = append(args.vec, "--"+byzantine.CfgActivationEpoch, strconv.FormatUint(uint64(epoch), 10))
+	args.vec = append(args.vec, argument{false, []string{"--" + byzantine.CfgActivationEpoch, strconv.FormatUint(uint64(epoch), 10)}})
 	return args
 }
 
 func (args *argBuilder) byzantineRuntimeID(runtimeID common.Namespace) *argBuilder {
-	args.vec = append(args.vec, "--"+byzantine.CfgRuntimeID, runtimeID.String())
+	args.vec = append(args.vec, argument{false, []string{"--" + byzantine.CfgRuntimeID, runtimeID.String()}})
 	return args
 }
 
+func (args *argBuilder) merge() []string {
+	output := []string{}
+	shipped := map[string]struct{}{}
+	multiValued := map[string][][]string{}
+
+	slicesEqual := func(s1, s2 []string) bool {
+		if len(s1) != len(s2) {
+			return false
+		}
+		for i := range s1 {
+			if s1[i] != s2[i] {
+				return false
+			}
+		}
+		return true
+	}
+
+	for _, arg := range args.vec {
+		leader := arg.items[0]
+		if arg.multiValued {
+			ok := true
+			for _, el := range multiValued[leader] {
+				if slicesEqual(el, arg.items) {
+					ok = false
+					break
+				}
+			}
+			if ok {
+				output = append(output, arg.items...)
+				multiValued[leader] = append(multiValued[leader], arg.items)
+			}
+		}
+		if _, ok := shipped[leader]; !ok && !arg.multiValued {
```

Comment:
Ah although the way the "MultiHost" test is implemented it looks like the values will get overridden (even with different values).

Reply:
It's not really clear what to do here. Every feature needs to provide all options, even non-multivalued ones, since every feature needs to be able to fully configure a node by itself. On the other hand, it's perfectly legitimate to have multiple occurrences of a single value option in the argument list if there are multiple features per node, so some sort of merging is definitely needed.
One thing that could make sense would be to panic if their values differ.